### PR TITLE
8.0 improve pr291 stock quant partner info

### PR DESCRIPTION
**Pull request by gurneyalex**

### CODE REVIEW

**Comment by gurneyalex** [comment]
fix #291

**Comment by pedrobaeza** [comment]
@gurneyalex No need of this large change. Just change the line to `move = quant.history_ids[-1:]`. `False` values are handled automatically at the start of the compute method (All fields are set to their default value before the compute method is called, so no need to reassign to `False`).

**Comment by gurneyalex** [comment]
@pedrobaeza no this does not work : ensure_one() crashes when you test move.location_dest_id.usage if move is an empty recordset.

**Comment by pedrobaeza** [comment]
I don't get it... let me put the whole compute method:
```python
     @api.depends('location_id', 'history_ids')
     def _compute_partner_id(self):
        for quant in self.filtered(
                lambda q: q.location_id.usage in ['customer', 'supplier']):
            move = quant.history_ids[-1:]
            quant.dest_partner_id =\
                move.picking_id.partner_id.commercial_partner_id
```

File: stock_quant_partner_info/models/stock_quant.py [hunk 1]
```diff
@@ -14,8 +14,12 @@ class StockQuant(models.Model):
 
     @api.depends('location_id', 'history_ids')
```
Comment (rousseldenis) [inline stock_quant_partner_info/models/stock_quant.py]: @gurneyalex You can improve your depends with 

```python
'history_ids.picking_id.partner_id.commercial_partner_id', 'history_ids.location_id.usage'
```
Avoiding fetching data inside function

File: stock_quant_partner_info/models/stock_quant.py [hunk 2]
```diff
@@ -14,8 +14,12 @@ class StockQuant(models.Model):
...
-                lambda q: q.location_id.usage in ['customer', 'supplier']):
-            move = quant.history_ids[:-1]
-            quant.dest_partner_id =\
-                move.picking_id.partner_id.commercial_partner_id
+        for quant in self:
+            moves = quant.history_ids.sorted(key=lambda m: m.date)
+            dest_partner = False
+            if moves:
+                move = moves[-1]
+                if move.location_id in ('customer', 'supplier'):
```
Comment (rousseldenis) [inline stock_quant_partner_info/models/stock_quant.py]:
```python
move.location_id.usage
```

File: stock_quant_partner_info/models/stock_quant.py [hunk 3]
```diff
@@ -14,8 +14,12 @@ class StockQuant(models.Model):
...
-        for quant in self.filtered(
-                lambda q: q.location_id.usage in ['customer', 'supplier']):
-            move = quant.history_ids[:-1]
-            quant.dest_partner_id =\
-                move.picking_id.partner_id.commercial_partner_id
+        for quant in self:
+            moves = quant.history_ids.sorted(key=lambda m: m.date)
+            dest_partner = False
+            if moves:
+                move = moves[-1]
```
Comment (rousseldenis) [inline stock_quant_partner_info/models/stock_quant.py:23]: What's the aim of this ?

**Review by rousseldenis — CHANGES_REQUESTED** [review]

File: stock_quant_partner_info/models/stock_quant.py (hunk 3 above)
Comment (gurneyalex) [reply stock_quant_partner_info/models/stock_quant.py:23]: @rousseldenis The original code would not update the dest_partner of a quant which has been returned from customer to stock for RMA for instance as it only operated on quants in customers or suppliers.

**Comment by pedrobaeza** [comment]
@gurneyalex please check my code. It's bulletproof and gets what you want.

**Comment by rousseldenis** [comment]
@gurneyalex What's the status of this ?

**Comment by rousseldenis** [comment]
@gurneyalex

File: stock_quant_partner_info/models/stock_quant.py (hunk 2 above)
Comment (rousseldenis) [reply stock_quant_partner_info/models/stock_quant.py]: @gurneyalex Maybe just fix this to get it merged ?

**Comment by gurneyalex** [comment]
closing, outdated.

**Pull request closed without merging**